packages/console/src/pages/ApplicationDetails/index.tsx

```diff
@@ -69,94 +71,97 @@ const ApplicationDetails = () => {
     reset(data);
   }, [data, reset]);
 
-  const {
-    field: { value: redirectUris, onChange: onRedirectUriChange },
-  } = useController({
-    control,
-    name: 'oidcClientMetadata.redirectUris',
-    defaultValue: [],
-  });
-
-  const {
-    field: { value: postSignOutRedirectUris, onChange: onPostSignOutRedirectUriChange },
-  } = useController({
-    control,
-    name: 'oidcClientMetadata.postLogoutRedirectUris',
-    defaultValue: [],
-  });
-
   const onSubmit = handleSubmit(async (formData) => {
     if (!data || isSubmitting) {
       return;
     }
 
     const updatedApplication = await api
-      .patch(`/api/applications/${data.id}`, { json: formData })
+      .patch(`/api/applications/${data.id}`, {
+        json: {
+          ...formData,
+          oidcClientMetadata: {
+            ...formData.oidcClientMetadata,
+            redirectUris: formData.oidcClientMetadata.redirectUris.filter(Boolean),
+            postLogoutRedirectUris:
+              formData.oidcClientMetadata.postLogoutRedirectUris.filter(Boolean),
+          },
+        },
+      })
```

Comment:
what about the api exceptions?

Reply:
It will show a toast by default. this is ensured by our `api` hook.

Reply:
LOG-2076 - Core: Application Redirect Uri Validation